src/test/java/org/example/integration/ApplicationIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+                "admin@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        Client client = APP.client();
+        LoginRequest loginRequest = new LoginRequest(
+                "admin@kainos.com",
+                "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6"
+
+        );
+        String token = "Bearer " + client
+                .target(System.getenv("API_URL")+"auth/login")
+                .request()
+                .post(Entity.json(loginRequest))
+                .readEntity(String.class);
+
+
+        Response response = client
+                .target(System.getenv("API_URL")+"apply")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .post(Entity.json(applicationRequest));
+
+        Assertions.assertEquals(201, response.getStatus());
+
+        Response deleteResponse = client
+                .target(System.getenv("API_URL")+"apply/delete/"+applicationRequest.getRoleId())
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .post(Entity.json(applicationRequest.getEmail()));
+
+        Assertions.assertEquals(204, deleteResponse.getStatus());
+
+    }
+
+    @Test
+    public void createApplication_shouldReturn500_whenCreatingExistingApplication() {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        Client client = APP.client();
+
+        LoginRequest loginRequest = new LoginRequest(
+                "admin@kainos.com",
+                "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6"
+
+        );
+        String token = "Bearer " + client
+                .target(System.getenv("API_URL")+"auth/login")
+                .request()
+                .post(Entity.json(loginRequest))
+                .readEntity(String.class);
+
+
+        Response response = client
+                .target(System.getenv("API_URL")+"apply")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .post(Entity.json(applicationRequest));
+
+        Assertions.assertEquals(500, response.getStatus());
+    }
+
+    @Test
+    public void createApplication_shouldReturn400_whenCreatingApplicationWithInvalidJobRole() {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                -1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
Look at previous comments, remove URL as you are exposing it to the world. The penguins are moving to Birmingham as they believe they can spam all the CVs with all the penguins in the world and take all the jobs available, even 100,027.